File: app/src/main/java/com/tohami/egyptinnovate/app/localization/Language.java
package com.tohami.egyptinnovate.app.localization;

import com.tohami.egyptinnovate.utilities.Constants;

import org.jetbrains.annotations.NotNull;

public enum Language {
    ENGLLSH(Constants.LocaleCode.ENGLISH), ARABIC(Constants.LocaleCode.ARABIC) , DEFAULT(Constants.LocaleCode.DEFAULT_LANGUAGE);

    String languageCode ;

    Language(String code) {
        this.languageCode = code ;
    }

    public String getLanguageCode() {
        return languageCode;
    }

    @NotNull
    public static Language fromLocaleCode(String value) {
        if (value == null) {
            return DEFAULT;
        }
        switch (value) {
            case Constants.LocaleCode.ARABIC:
                return ARABIC;

            case Constants.LocaleCode.ENGLISH:
                return ENGLLSH;

            default:
                return DEFAULT ;
        }

    }
}

File: app/src/main/java/com/tohami/egyptinnovate/app/localization/Language.kt
package com.tohami.egyptinnovate.app.localization

import com.tohami.egyptinnovate.utilities.Constants

enum class Language (val code: String) {
    ENGLLSH(Constants.LocaleCode.ENGLISH),
    ARABIC(Constants.LocaleCode.ARABIC),
    DEFAULT(Constants.LocaleCode.DEFAULT_LANGUAGE);

    companion object {

        fun fromLocaleCode(value: String?): Language {
            if (value == null) {
                return DEFAULT
            }
            return when (value) {
                Constants.LocaleCode.ARABIC -> ARABIC

                Constants.LocaleCode.ENGLISH -> ENGLLSH

                else -> DEFAULT
            }

        }
    }
}
